### PR TITLE
modernize packaging using pyproject.toml

### PR DESCRIPTION
**Pull request by dulmandakh**
This PR is modernizing packaging using pyproject.toml, a recommended way for Python projects. Please see https://packaging.python.org/en/latest/guides/modernize-setup-py-project/.

I also added py38-build env in tox, so with every change it'll package and check if it's distributable using twine.

### CODE REVIEW

**Comment by dulmandakh** [comment]
Tests are failing due to Docker Hub rate limit.

File: tox.ini [hunk 1]
```diff
@@ -19,16 +19,18 @@ TARGET =
   wand: wand
   dbm: dbm
   qa: qa
+  build: build
```
Comment (claudep) [inline tox.ini:22]: I'm not used to add the build step inside tox.ini. Is it really needed?

Comment (dulmandakh) [reply tox.ini:22]: Just to make sure that it's buildable 🤷‍♂️ But not that necessary for python. I can remove it.

File: tox.ini [hunk 2]
```diff
@@ -53,7 +53,7 @@ setenv =
   pgmagick: DJANGO_SETTINGS_MODULE=tests.settings.pgmagick
   dbm: DJANGO_SETTINGS_MODULE=tests.settings.dbm
 commands =
-  python -m pytest -rw --cov-append --cov-config setup.cfg --cov sorl --cov-report=xml {posargs:}
+  python -m pytest -rw --cov-append --cov sorl --cov-report=xml {posargs:}
```
Comment (claudep) [inline tox.ini:56]: Could you check PR #755, as I think we don't need pytest at all ?

Comment (dulmandakh) [reply tox.ini:56]: I can update this PR once #755 merged

**Comment by dulmandakh** [comment]
@claudep rebased on master, including your pytest changes.

**Review by claudep — APPROVED** [review]
:+1:

**Pull request merged**